### PR TITLE
Config flag to stop template from local resolution

### PR DESCRIPTION
**Pull request by mixonic**
- Add `disableLocalResolutionForTypes` to the config, permits a type to block being locally resolved in a given collection.

*edit*: This summary is out of date. See https://github.com/glimmerjs/glimmer-resolver/pull/17#issuecomment-312459450

### CODE REVIEW

**Comment by mixonic** [comment]
Adds two tests:

* https://github.com/glimmerjs/glimmer-resolver/blob/90a4079f8d4094df8c63dcde04ad8437ce1bbd89/test/resolution-order-test.ts#L69
* https://github.com/glimmerjs/glimmer-resolver/blob/90a4079f8d4094df8c63dcde04ad8437ce1bbd89/test/resolution-order-test.ts#L148

File: test/resolution-order-test.ts [hunk 1]
```diff
@@ -46,13 +46,43 @@ test('Identifies `template` from `component:/app/components/my-form`', function(
...
       template: {
-      },
-      component: {
         definitiveCollection: 'components'
-      }
+      },
+      component: {}
     },
     collections: {
+      routes: {
```
Comment (dgeb) [inline test/resolution-order-test.ts]: shouldn't we add `types` in the `routes` collection?

Comment (mixonic) [inline test/resolution-order-test.ts]: yes thank you, adding this.

**Comment by mixonic** [comment]
Ok, the approach her has changed. Two semantic differences land here:

* First, if you resolve with only a type (`'type'`, not `'type:name'`) then no resolution fallback steps will run. The type will be looked for only in namespace and collection of the referrer. This is useful for local lookup, and to force the collection of a global lookup (From Ember we call `resolve('template', 'route:/app/routes/my-page')` to force a template lookup in the routes collection).
* Second, local lookup will only occur for a type if the collection is that type's `definitiveType`. In Ember this ensures `resolve('template:my-page', 'template:/app/routes/my-form')` does not resolve a module at `template:/app/routes/my-form/my-page`.

**Comment by mixonic** [comment]
One reason you might call this less than ideal is that `resolve('template', 'route:/app/routes/my-page')` implies you must have a second type (here `route`) in the `routes` collection in order to force lookup to a `template` in that collection. It might be better if `resolve('template', ':/app/routes/my-page')` without a type on the referrer was permitted.

**Pull request merged**